camel/utils/token_counting.py

```diff
@@ -179,8 +179,23 @@ def count_tokens_from_messages(self, messages: List[OpenAIMessage]) -> int:
         Returns:
             int: Number of tokens in the messages.
         """
-        prompt = messages_to_prompt(messages, self.model_type)
-        input_ids = self.tokenizer(prompt).input_ids
+        if self.model_type == ModelType.QWEN_2:
+            text = self.tokenizer.apply_chat_template(
+                messages, tokenize=False, add_generation_prompt=True
+            )
+            input_ids = self.tokenizer([text], return_tensors="pt").input_ids
+        elif self.model_type == ModelType.GLM_4_OPEN_SOURCE:
+            inputs = self.tokenizer.apply_chat_template(
+                messages,
+                add_generation_prompt=True,
+                tokenize=True,
+                return_tensors="pt",
+                return_dict=True,
+            )
+            input_ids = inputs["input_ids"]
+        else:
+            prompt = messages_to_prompt(messages, self.model_type)
+            input_ids = self.tokenizer(prompt).input_ids
```

Comment:
Hey @zhiyu-01 ,could you help me understand why the set for token counter here is different for `QWEN_2` and `GLM_4_OPEN_SOURCE`? Any reference for this? Thanks in advance!

Reply:
https://huggingface.co/THUDM/glm-4-9b-chat
https://huggingface.co/Qwen/Qwen2-7B-Instruct

I found these two settings from their huggingface repo.

Reply:
Thanks for the answer, actually I think the 2 code blocks are doing the same thing, maybe we can unify the code

Reply:
I have run the models, and set the str prompt correspondingly.
qwen:
![qwen](https://github.com/camel-ai/camel/assets/121875294/bc529040-42d5-4fb0-9e5d-e0fc6682d74b)
glm:
![glm](https://github.com/camel-ai/camel/assets/121875294/e8128edc-c8b0-4186-8d51-da43825c93d2)